mono_repo/lib/src/utils.dart

```diff
@@ -125,12 +131,41 @@ Map<String, MonoConfig> getMonoConfigs(
 
 class UserException implements Exception {
   final String message;
+  final String details;
 
-  UserException(this.message);
+  UserException(this.message, {this.details});
 
   @override
   String toString() => 'UserException: $message';
 }
 
 String encodeJson(Object input) =>
     const JsonEncoder.withIndent(' ').convert(input);
+
+String prettyPrintCheckedFromJsonException(CheckedFromJsonException err) {
```

Comment:
should we just create a package for this at this point? or add a yaml dependency to source_gen?